client/components/result-status-icon/icon.directive.js

```diff
@@ -0,0 +1,21 @@
+angular.module('probedock.resultStatusIcon').directive('resultStatusIcon', function() {
+  return {
+    restrict: 'E',
+    controller: 'ResultStatusIconCtrl',
+    templateUrl: '/templates/components/result-status-icon/icon.template.html',
+    scope: {
+      result: '='
+    }
+  };
+}).controller('ResultStatusIconCtrl', function($scope) {
+  $scope.ready = false;
+
+  $scope.$watch('result', function (value) {
+    if (value) {
+      $scope.passed = ($scope.result.passed || $scope.result.passing) && $scope.result.active;
```

Comment:
If this directive can display the result for a test or one of the results of that test, then it should be named `test-status-icon` and have two possible arguments: either a test or a result. Passing a test in an argument named `result` is quite counter-intuitive.

Or if it's only used for results, then you can remove the `passing` fallback.